apps/gdalalg_vector_update.cpp

```diff
@@ -0,0 +1,350 @@
+/******************************************************************************
+ *
+ * Project:  GDAL
+ * Purpose:  "update" step of "vector pipeline"
+ * Author:   Even Rouault <even dot rouault at spatialys.com>
+ *
+ ******************************************************************************
+ * Copyright (c) 2025, Even Rouault <even dot rouault at spatialys.com>
+ *
+ * SPDX-License-Identifier: MIT
+ ****************************************************************************/
+
+#include "gdalalg_vector_update.h"
+
+#include "ogr_p.h"
+#include "ogrsf_frmts.h"
+
+//! @cond Doxygen_Suppress
+
+#ifndef _
+#define _(x) (x)
+#endif
+
+/************************************************************************/
+/*        GDALVectorUpdateAlgorithm::GDALVectorUpdateAlgorithm()        */
+/************************************************************************/
+
+GDALVectorUpdateAlgorithm::GDALVectorUpdateAlgorithm(bool standaloneStep)
+    : GDALVectorPipelineStepAlgorithm(NAME, DESCRIPTION, HELP_URL,
+                                      ConstructorOptions()
+                                          .SetStandaloneStep(standaloneStep)
+                                          .SetInputDatasetMaxCount(1)
+                                          .SetAddInputLayerNameArgument(false)
+                                          .SetAddDefaultArguments(false))
+{
+    if (standaloneStep)
+    {
+        AddVectorInputArgs(false);
+    }
+    {
+        auto &layerArg = AddArg(GDAL_ARG_NAME_INPUT_LAYER, 0,
+                                _("Input layer name"), &m_inputLayerNames)
+                             .SetMaxCount(1);
+        auto inputArg = GetArg(GDAL_ARG_NAME_INPUT);
+        if (inputArg)
+            SetAutoCompleteFunctionForLayerName(layerArg, *inputArg);
+    }
+
+    AddProgressArg();
+
+    AddOutputDatasetArg(&m_outputDataset, GDAL_OF_VECTOR)
+        .SetDatasetInputFlags(GADV_NAME | GADV_OBJECT);
+    AddOutputOpenOptionsArg(&m_outputOpenOptions);
+    AddArg(GDAL_ARG_NAME_OUTPUT_LAYER, 0, _("Output layer name"),
```

Comment:
Is it possible to use `GDALAlgorithm::AddOutputLayerNameArg()` here ?

Reply:
done